ddtrace/appsec/_iast/_langchain.py

```diff
@@ -222,6 +218,23 @@ def _get_tainted_source_from_chat_prompt_value(chat_prompt_value):
     return None
 
 
+def _get_text_attribute_for_generation(gen):
+    text_attr = None
+    if hasattr(gen, "_text"):
```

Comment:
if it is possible to check langchain version here instead, it might be more performant

e.g. at import/patch time extract the langchain version into a global variable, and then use here.

"probably" more performant.

Reply:
this is not a blocking comment, just a random thought, resolve and ignore if you want.

Reply:
Thanks, I'll look into how to simplify runtime callbacks by selecting specialized versions at patch time. It's a good point. I'm not excessively worried about this particular one, since it's roughly one extra `hasattr` call worst case. But it might be possible to streamline the callbacks overall.

Reply:
yeah, the global lookup/comparsion vs a `hasattr`, not 100% sure myself, which is why 🤷🏻 


can handle as a follow-up if you wanted.